crates/loam-cli/examples/soroban/nft/src/nft.rs

```diff
@@ -0,0 +1,118 @@
+use loam_sdk::{
+    soroban_sdk::{self, contracttype, env, Address, Bytes, Lazy, Map, Vec},
+    IntoKey,
+};
+
+use crate::subcontract::{IsInitable, IsNonFungible};
+
+#[contracttype]
+#[derive(IntoKey)]
+pub struct MyNonFungibleToken {
+    admin: Address,
+    name: Bytes,
+    total_count: u128,
+    owners_to_nft_ids: Map<Address, Map<u128, ()>>, // the owner's collection
+    nft_ids_to_owners: Map<u128, Address>,
+    nft_ids_to_metadata: Map<u128, Bytes>,
+}
+
+impl MyNonFungibleToken {
+    #[must_use]
+    pub fn new(admin: Address, name: Bytes) -> Self {
+        MyNonFungibleToken {
+            admin,
+            name,
+            total_count: 0,
+            owners_to_nft_ids: Map::new(env()),
+            nft_ids_to_owners: Map::new(env()),
+            nft_ids_to_metadata: Map::new(env()),
+        }
+    }
+}
+
+impl Default for MyNonFungibleToken {
+    fn default() -> Self {
+        MyNonFungibleToken::new(env().current_contract_address(), Bytes::new(env()))
```

Comment:
Wait, we are setting the admin to the contract itself?

Reply:
Why even bother implementing `Default`?

Reply:
Is it something to do with the `&mut self` conversation [above](https://github.com/loambuild/loam/pull/91#pullrequestreview-2151758328)?

Reply:
Right now `Default` is required by the macro

Reply:
it's a nonsensical default. i don't like it. Will this fix it?

- #92